### DISCUSSION TITLE
Multiplayer server not connecting for some Russian ISPs

### DISCUSSION THREAD

**Discussion by Misadov** [ppy/osu#19931] [Q&A]
### Type

Game behaviour

### Bug description

Multiplayer rooms don't load. Infinity loading (I waited for 15 mins, before bug loading took 5 - 10 seconds)

### Screenshots or videos

https://user-images.githubusercontent.com/81818882/173930725-4f320456-6c51-4c88-8b33-8e434e77c43b.mp4

### Version

2022.615.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/8913008/database.log)
[network.log](https://github.com/ppy/osu/files/8913009/network.log)
[performance.log](https://github.com/ppy/osu/files/8913010/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8913011/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8913012/updater.log)

**Comment by bdach** [comment]
I cannot reproduce this failure. The logs contain an error message in Russian, but I'm pretty sure the English correspondent of it is:

```
A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
```

Other users [have already](https://github.com/ppy/osu/discussions/18565#discussioncomment-2935703) [been reporting](https://github.com/ppy/osu/discussions/18565#discussioncomment-2940272) similar. There's a strong chance that hitting this is highly dependent on the user's geographical location.

**Comment by Misadov** [comment]
I tried to use VPN (still allowed in Russia) and lobbies are working fine for me. Either ppy blocked russians or hosting provider set restrictions because of the current situation.

**Comment by peppy** [comment]
No restrictions have been set at our (or our host’s) end.

We switched cloudflare off to get better websocket reliability. We will add some fallbacks in the coming future to handle cases like yours.

You are welcome to use a VPN for now.

**Comment by xDololow** [comment]
> I tried to use VPN (still allowed in Russia) and lobbies are working fine for me. Either ppy blocked russians or hosting provider set restrictions because of the current situation.

what ISP do you use?

edit: I could reproduce this issue on my side. With VPN it works fine, without loong loading.

**Comment by Misadov** [comment]
> > I tried to use VPN (still allowed in Russia) and lobbies are working fine for me. Either ppy blocked russians or hosting provider set restrictions because of the current situation.
> 
> what ISP do you use?
> 
> edit: I could reproduce this issue on my side. With VPN it works fine, without loong loading.

My ISP is MGTS OAO (https://2ip.io/isp/OJS+Moscow+city+telephone+network/)

> You are welcome to use a VPN for now.

Okay, thanks.

**Comment by xDololow** [comment]
> My ISP is MGTS OAO

I see. I use local ISP, but it uses Megafon backbone lines.

**Comment by Loreos7** [comment]
My ISP is MGTS OAO as well
https://ping-admin.com/free_test/
Problem is held only in Russia

**Comment by BKVad1m** [comment]
> No restrictions have been set at our (or our host’s) end.
> 
> We switched cloudflare off to get better websocket reliability. We will add some fallbacks in the coming future to handle cases like yours.
> 
> You are welcome to use a VPN for now.

If using a VPN gets banned another site when using VPN. How to use VPN osu!lazer application only?
I used VPN and discord not working, I uninstalled VPN and discord is working.

**Comment by stanriders** [comment]
Same issue, SkyNet ISP, Saint-Petersburg, Russia. Using a VPN works but is obviously far from ideal.  

Pings to `spectator.ppy.sh` work, tracert more or less works. Fiddler shows time out, wireshark doesn't really seem to show anything useful.  
![image](https://user-images.githubusercontent.com/8269193/175422906-e49de87e-0d20-4261-845f-4dc21a35f553.png)  
![image](https://user-images.githubusercontent.com/8269193/175422809-cab541ed-8487-461e-a8f6-750e08419300.png)  
![image](https://user-images.githubusercontent.com/8269193/175422628-4c6ded96-af98-4b19-bf6d-22c58269fc00.png)  
![image](https://user-images.githubusercontent.com/8269193/175422640-89d82937-13f0-41f4-9dad-967a80ef44ce.png)

**Comment by peppy** [comment]
@stanriders does the page load in your browser (https://spectator.ppy.sh)?

also can you confirm that the wireshark screenshot is showing bidirectional traffic (but it's only flowing in one direction)?

**Comment by stanriders** [comment]
@peppy the page does not load and times out, Wireshark had no incoming traffic logged from it at all.

**Comment by peppy** [comment]
The weirdest part is that you can ping but not get an http response. This sounds like some custom firewall rules, potentially at your ISP's end. Are you able to try and find some other DO hosted sites and see if you have similar issues?

**Comment by stanriders** [comment]
I'm not really sure how I can find any DO sites to test, especially considering most of the internet is going through cloudflare these days. If you have any examples I can run whatever you need

**Comment by SyrokVincent** [comment]
If that can help, multiplayer rooms don't load for me either, I'm in France and my ISP is "Free SAS"

https://spectator.ppy.sh/ gives me white screen
https://spectator2.ppy.sh/ - connection has timed out, can't find the page

**Comment by stanriders** [comment]
Speaking of, `spectator2` doesn't resolve for me at all

**Comment by smoogipoo** [comment]
`spectator2` doesn't exist anymore.

**Comment by pandulce08** [comment]
Same error here, i'm from argentina.

**Comment by peppy** [comment]
> Same error here, i'm from argentina.

Do you get an error or a white screen when visiting https://spectator.ppy.sh?

**Comment by xDololow** [comment]
Right now it gives me 404 Not Found.

**Comment by peppy** [comment]
Can you please take a screenshot of what you're seeing?

**Comment by xDololow** [comment]
> Can you please take a screenshot of what you're seeing?

Chrome (says page not found)
![image](https://user-images.githubusercontent.com/26763912/176866006-3c8f1d89-dc46-4be0-befa-b66c4379a6d9.png)

Firefox
![image](https://user-images.githubusercontent.com/26763912/176866101-f741ecaa-d7ea-4192-8bd7-3388456922ae.png)

**Comment by peppy** [comment]
Are you able to test whether this site works correctly? https://www.piesocket.com/websocket-tester

It seems like you can see the server (404 response is correct) so my next theory is that somehow your ISP is blocking websocket connections.

**Comment by xDololow** [comment]
No problems on this one, works fine for me
![image](https://user-images.githubusercontent.com/26763912/176867199-5c7dae5e-6a3d-4130-ad0d-4d0b26ca177f.png)

**Comment by peppy** [comment]
And just to 100% confirm, osu! still doesn't work for you even with all the above results? Your web browser is able to connect to the server fine, so I can't think of a reason why osu! would be unable to. There has to be something more to this, especially since other users in different regions are experiencing the same thing.

Could you please provide the game's `network.log` once more?

**Comment by xDololow** [comment]
Actions I did:

1. Open osu!lazer
2. clicked to play -> multiplayer and waited a couple of minutes for it to connect (unsuccessful)
3. clicked back to main menu and then to playlists (loaded without a problem), opened random playlist and closed back to main menu
4. tried multiplayer once again, and it loaded after a few seconds.

[network.log](https://github.com/ppy/osu/files/9027958/network.log)

Seems like it can load, but not always (like 1 out of 10 tries and with random success timing🗿). System.Net.Sockets.SocketException (10060) returns `An attempt to establish a connection was unsuccessful because no response was received from the other computer in the required time, or an already established connection was broken due to an incorrect response from an already connected computer.`

**Comment by Misadov** [comment]
If it makes sense, traceroute path fails on the first jump (on router) with timeout, IDK why. 
![cmd_NtXJVv6tey](https://user-images.githubusercontent.com/81818882/176881186-ee1b5b9b-9585-414a-8927-b3b9b8d440b8.png)
(for example, there is screenshot of traceroute to site that is blocked in Russia)
![cmd_h0taiaKEgz](https://user-images.githubusercontent.com/81818882/176881200-cd8095cf-0cee-465e-b0fc-485f87c56138.png)

)

**Comment by peppy** [comment]
Traceroute failing on certain hops is not an issue, and to be expected. But what you are seeing looks like your ISP is routing your packets to a black hole.

**Comment by Loreos7** [comment]
For me now it looks like this 
![image](https://user-images.githubusercontent.com/86934170/176885333-c92d9368-1f7c-4e66-8385-4e7003e5a44c.png)

**Comment by stanriders** [comment]
> The weirdest part is that you can ping but not get an http response. This sounds like some custom firewall rules, potentially at your ISP's end. Are you able to try and find some other DO hosted sites and see if you have similar issues?

Would you mind opening a non-standart http port to see if that works? Something like 8123 (since 8080 might be blocked too) with a hello world

**Comment by pandulce08** [comment]
> > Same error here, i'm from argentina.
> 
> Do you get an error or a white screen when visiting https://spectator.ppy.sh?

I get a white screen on firefox and an error on chromium.

![12](https://user-images.githubusercontent.com/108196603/176892869-989dc3bc-11c4-4146-b84a-f73edadfe4d7.png)


![13](https://user-images.githubusercontent.com/108196603/176892876-72f0d334-bda1-4579-90cb-82c8e19825d2.png)

**Comment by Misadov** [comment]
I fixed the problem by calling the ISP and asking them to troubleshoot connection to spectator.ppy.sh. Now rooms loading and whole multiplayer is working fine for me.

**Comment by peppy** [comment]
Did they provide any further information about what they did to resolve this issue? If so that would be very helpful to us.

**Comment by Loreos7** [comment]
I have the same ISP user Misadov has but unfortunately it doesnt work for me

**Comment by Misadov** [comment]
> Did they provide any further information about what they did to resolve this issue? If so that would be very helpful to us.

Unfortunately no, just called and they said the tech guy didn't write comments about the problem. But no one came to my house and did not fix anything, so I can assume that the problem was on their side.
I can only add that they were only able to fix the problem when I as they said left session initialized (osu was opened for 3-4 hours at the multiplayer loading screen).



> I have the same ISP user Misadov has but unfortunately it doesnt work for me

I can only advise you to try calling again. You may not have been assigned to tech support line, but left on customer support line. You can contact me in discord Misadov#3693, we can talk in russian and I will try to help you.

**Comment by peppy** [comment]
@Loreos7 if you do contact support, please ask for more details. also mention that you are in contact with the developers of the service and request any details which could help us potentially fix this at our end.

**Comment by Loreos7** [comment]
> @Loreos7 if you do contact support, please ask for more details. also mention that you are in contact with the developers of the service and request any details which could help us potentially fix this at our end.

Sure, I'll do

**Comment by Misadov** [comment]
I called ISP again (same issue came again) and talked with some main tech staff (I said that I in contact with developers). They said that some routes are blocked and your domain (spectator.ppy.sh) can be pinged from specific locations. I said that there is no blocking from your side, they said that this problem is global and from other ISP domain also can't be pinged. That's all information I got. 

Also, main tech staff said that there is no information how issue was fixed before.

**Comment by TheAirBlow** [comment]
I have the same problem. My provider is [OJSC Ufanet](https://2ip.io/isp/OJSC+Ufanet/), sometimes I can visit some blocked websites. 
I am an arch user too, and osu! version is `2022.723.0-lazer`.

### Troubleshooting
#### Websocket connection test https://github.com/ppy/osu/issues/18721#issuecomment-1172137083
URL: `https://www.piesocket.com/websocket-tester`
![image](https://user-images.githubusercontent.com/68467762/180666297-46183e45-a110-44c0-839e-0bfec541ada7.png)
For some unknown reason I was often getting timeouts instead of 404s without VPN or Tor.
#### Trace routing https://github.com/ppy/osu/issues/18721#issuecomment-1164931765
Used `traceroute` for this purpose.
```
traceroute to spectator.ppy.sh (167.172.8.103), 30 hops max, 60 byte packets
 1  _gateway (192.168.0.1)  2.866 ms  3.225 ms  3.914 ms
 2  100.92.160.1 (100.92.160.1)  15.662 ms  15.978 ms  16.323 ms
 3  10.1.132.2 (10.1.132.2)  16.911 ms  17.287 ms  17.681 ms
 4  10.1.132.17 (10.1.132.17)  18.416 ms  18.635 ms  18.992 ms
 5  10.1.132.57 (10.1.132.57)  21.354 ms  21.687 ms  22.015 ms
 6  10.0.19.153 (10.0.19.153)  19.708 ms  17.437 ms  17.475 ms
 7  79.140.16.179 (79.140.16.179)  17.294 ms 79.140.16.195 (79.140.16.195)  7.469 ms 79.140.16.179 (79.140.16.179)  12.148 ms
 8  79.140.16.193 (79.140.16.193)  12.966 ms  14.235 ms  14.192 ms
 9  10.2.168.1 (10.2.168.1)  15.518 ms  16.085 ms  16.963 ms
10  10.1.245.8 (10.1.245.8)  14.148 ms  14.444 ms  14.816 ms
11  10.1.245.130 (10.1.245.130)  16.358 ms  17.318 ms  17.723 ms
12  212.46.219.109 (212.46.219.109)  18.555 ms  18.650 ms  6.648 ms
13  * * *
14  * * *
15  ae1.3505.edge9.SanJose1.level3.net (4.69.219.61)  289.538 ms  289.109 ms  289.009 ms
16  DIGITAL-OCE.edge9.SanJose1.Level3.net (4.71.117.218)  289.340 ms * 4.7.18.10 (4.7.18.10)  289.814 ms
```

**Comment by TheAirBlow** [comment]
It works for now for some reason
P.S. Stopped working again as of 08.14.2022

**Comment by Loreos7** [comment]
@peppy 
I've noticed a change with the spectator and after the last update multiplayer has gone accessible!
![image](https://user-images.githubusercontent.com/86934170/185900661-6a09e061-5f43-4898-9924-57dfa8aea6d9.png)

**Comment by peppy** [comment]
That's great, but we didn't change anything from our end. It may have started on a different digitalocean IP in our cluster, which would assert that certain IPs are being blocked by your ISP even more so.

**Comment by Loreos7** [comment]
@xDololow @Vad1mCSO2 @stanriders @pandulce08 @SyrokVincent @TheAirBlow 
Could you confirm accessibility or its still unavailable?

**Comment by stanriders** [comment]
Seems to be working for me now

**Comment by TheAirBlow** [comment]
> @xDololow @Vad1mCSO2 @stanriders @pandulce08 @SyrokVincent @TheAirBlow 
> Could you confirm accessibility or its still unavailable?

It works now, didn't work yesterday lol

**Comment by Misadov** [comment]
Works fine for me as well

**Comment by xDololow** [comment]
> @xDololow @Vad1mCSO2 @stanriders @pandulce08 @SyrokVincent @TheAirBlow Could you confirm accessibility or its still unavailable?

Works for me.

**Comment by pandulce08** [comment]
> @xDololow @Vad1mCSO2 @stanriders @pandulce08 @SyrokVincent @TheAirBlow Could you confirm accessibility or its still unavailable?

Still doesn't work for me

**Comment by 6yntar05** [comment]
Worked yesterday, stopped again today

**Reply by Loreos7** [reply]
Same for me again

**Reply by 6yntar05** [reply]
> Same for me again

After single traceroute its works again
I don't know what's going on with routings
._.

**Comment by peppy** [comment]
You're going to need to take this up with your ISP. It's 100% not our side (or our server's side).

**Comment by BKVad1m** [comment]
![image](https://user-images.githubusercontent.com/76803825/187348193-86e84daa-6204-4557-bed0-2bfd56bc06be.png)
I got an error
